Solution UI: improve code

carlito-scality opened the pull request:
**Component**: Solution UI

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 
- Address all the remaining comments of the PR #1783

**Summary**:

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: https://github.com/scality/metalk8s/issues/1821

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

carlito-scality committed:
Solution: rewording and fix typos

bert-e commented:
# Hello carlito-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

carlito-scality committed:
Replace "list" methode by "read" methode
- listNamespacedDeployment
- listNamespacedServiceAccount
- listNamespacedRole
- listNamespacedRoleBinding
- listConfigMapForAllNamespaces

carlito-scality committed:
Set replicas default value as a number

carlito-scality committed:
Rename duck constants to make it simpler

carlito-scality committed:
Replace listNamespace by readNamespace

carlito-scality committed:
Merge branch 'development/2.4' into feature/solution-UI-cleaning-code

* development/2.4: (6 commits)
  ui/cypress Update Cypress binary and package link
  ...

gdemonet commented on examples/metalk8s-solution-example/ui/src/ducks/app/clockServer.js:
```diff
@@ -85,18 +85,18 @@ export function* createClockServer({ payload }) {
   const { name, environment, timezone, version } = payload;
   const body = {
-    apiVersion: 'example-solution.metalk8s.scality.com/v1alpha1',
+    apiVersion: `${SOLUTION_NAME}.metalk8s.scality.com/v1alpha1`,
```
Conceptually, this is not portable to other Solutions. A better approach would define (in `constants.js`):

```javascript
const SOLUTION_API_GROUP = `${SOLUTION_NAME}.metalk8s.scality.com`
```

And you could use this value in here, only specifying `v1alpha1` for each CR (it could differ between them, so no need to attach it in `constants.js`).

gdemonet commented on examples/metalk8s-solution-example/ui/src/ducks/app/deployment.js:
```diff
@@ -10,22 +10,22 @@ import {
 
 // Sagas
-export function* fetchOpertorDeployments(namespaces) {
+export function* fetchOpertorDeployment(namespaces) {
```
Oops:

```suggestion
export function* fetchOperatorDeployment(namespaces) {
```

gdemonet reviewed (dismissed):
Few nitpicks, but thanks for fixing all these small details. I'll need to change a few things anyway, related to #1912, so I think it's safe to merge as-is.

carlito-scality committed:
Extract SOLUTION_GROUP to the constant.js

gdemonet commented on examples/metalk8s-solution-example/ui/src/constants.js:
```diff
@@ -5,6 +5,7 @@ export const LABEL_NAME = 'app.kubernetes.io/name';
 export const LABEL_PART_OF = 'app.kubernetes.io/part-of';
 export const LABEL_VERSION = 'app.kubernetes.io/version';
 export const SOLUTION_NAME = 'example-solution';
+export const SOLUTION_GROUP = `${SOLUTION_NAME}.metalk8s.scality.com`;
```
`API_GROUP`, not just `GROUP`, otherwise it's confusing

gdemonet commented on examples/metalk8s-solution-example/ui/src/ducks/app/deployment.js:
```diff
@@ -256,7 +256,7 @@ const roleBody = {
     },
     {
-      apiGroups: ['example-solution.metalk8s.scality.com'],
+      apiGroups: [`${SOLUTION_NAME}.metalk8s.scality.com`],
```
```suggestion
      apiGroups: [SOLUTION_API_GROUP],
```

carlito-scality committed:
Rename SOLUTION_GROUP to SOLUTION_API_GROUP to be clearer

carlito-scality commented:
@bert-e approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/5a79b1b8ac45ce685ded550b9d4ce201851400a7) did not succeed in branch feature/solution-UI-cleaning-code.




*The following options are set:* **approve**

carlito-scality committed:
Merge branch 'development/2.4' into feature/solution-UI-cleaning-code

* development/2.4: (3 commits)
  Test/upgrade: deactivate tests about upgrade
  ...

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye carlito-scality.
